src/js/components/TextInput/TextInput.js

```diff
@@ -480,6 +492,7 @@ const TextInput = forwardRef(
                     // placeholder only appears when there is no value
                     setValue(event.target.value);
                     setActiveSuggestionIndex(resetSuggestionIndex);
+                    setSuggestionsAtClose();
```

Comment:
Same question as before, why is `setSuggestionsAtClose()` being called here? And why is suggestionsAtClose being set to an empty value?

Reply:
Halo thanks for reviewing. My understanding is that `suggestionsAtClose` are needed in order to avoid re-opening the drop when select a value or blur out of the input and the drop closes. That means they are valid only for the current value in the input (since just focusing back, doesn't change the suggestions). As soon as we change the value though, we need to open the drop again and present the suggestions to the user. 
Check [this](https://storybook.grommet.io/?path=/story/input-textinput-custom-suggestions--custom-suggestions) for example: Type `alan` -> one suggestion comes up -> focus again, select the text and just replace it with `j`, drop won't open even though the suggestion (again just one) is completely different.

Reply:
I don't think we should be setting suggestionsAtClose to and empty value since that doesn't accurately reflect the intended value of the variable. Based on this comment:
```
// Handle scenarios where we have focus, the drop isn't showing,
// and the suggestions change. We don't want to open the drop if
// the drop has been closed by onEsc and the suggestions haven't
// changed. So, we remember the suggestions we are showing when
// the drop was closed and only re-open it when the suggestions
// subsequently change.
```
I think this is the intended use of suggestionsAtClose

Reply:
I'm sorry but maybe you haven't check the issue: https://github.com/grommet/grommet/issues/5426 and also I guess you haven't done what I wrote on my previous comment.
I will leave that PR here, if there is anyone else that wants to review it feel free, otherwise close it.
Chreerz

Reply:
Hi @novykh I did look at the issue and test using the Custom suggestions story you mentioned above. I was just thinking that setting the `suggestionsAtClose` variable might not be the best solution since it doesn't accurately reflect the intended value of the variable. Instead I think we could do something like this to achieve the desired behavior:
```javascript
onChange={
  readOnly
    ? undefined
    : (event) => {
        // when TextInput is not contained in a Form, no re-render
        // will come from this onChange and remove the placeholder
        // so we need to update state to ensure the styled
        // placeholder only appears when there is no value
        if (suggestions && focus && !showDrop) {
          openDrop();
        }
        setValue(event.target.value);
        setActiveSuggestionIndex(resetSuggestionIndex);
        if (onChange) onChange(event);
      }
}
onClick={() => {
  if (suggestions && !showDrop) {
    openDrop();
  }
}}
```

Reply:
Hey, ok maybe it's my personal opinion. But since this variable exists only "for when the drop will close" in order to hijack it let's say, and not re-open. By the moment the user changes the value, those suggestions are obsolete. I will try your suggestion, since it looks smaller and I don't want to change the way you've built things. But the data will be obsolete when checking the length of `suggestionsAtClose` vs `suggestions` since the length might be the same, but the suggestions will be different. Do you mind if I open later a new PR with your suggestion, and mention you for visibility?

Reply:
The suggestionsAtClose variable exists so that if the suggestions change and the TextInput still has focus the drop will reopen if it was closed with 'escape'. And yes you can open a new PR and mention me on it

Reply:
OK one month passed by, but I'm back.
@jcfilben I tried your solution, and after selecting a suggestion (by clicking it) the drop closes, thus when you click the input again, it re-opens. This is good, but it's not covering the whole issue here.
So to clarify here is an example where you use the TextInput for searching stuff:

1. type `a`
2. the suggestions drop opens with
```
all are great!
all are great?
```
3. you select one of them with your keyboard
4. the drop closes but you are still focused on the input.
5. you backspace and remove the punctuation character in the end
5.i. should show the 2 same suggestions again
6. nothing happens because when you selected there were 2 suggestion and again now there are 2 suggestions

So, given that people are using the suggestions prop in order to always display suggestions as the user types, I don't understand why aren't we supposed to reset the `suggestionsAtClose` after the user enters a new value - in the example above "removing a character".

You can check your thoughts against the test I added on this PR. Thank you for your time, but since I see that after all this debate there is not a common understanding on the fix needed I don't feel I can continue with this. You can go ahead and close the PR. Cheers